apps/dgst.c

```diff
@@ -413,6 +413,28 @@ int dgst_main(int argc, char **argv)
     return ret;
 }
 
+const char * newline_escape_filename(const char *file, int * backslash)
+{
+    int i = 0, length = strlen(file);
+    char *file_cpy = app_malloc(length + 3, file);
```

Comment:
This assumes that the newline will only appear once in the filename. I suggest either two passes over the file name, or an implementation that uses `OPENSSL_realloc`

Reply:
Two new lines, but I agree.

Reply:
Great suggestion.  I was unsure if 2 passes would be acceptable.  I will refactor with this in mind to calculate the total amount of newlines.